### ISSUE TITLE
Using scipy fsolve or other solvers (e.g. GEKKO) in combination with backpropagation models

### ISSUE THREAD

**Issue by cevheck** [tensorflow/tensorflow#53805]
I was wondering if there was any progress or existing methodologies to use existing solvers such as GEKKO to find/update certain values in your network.
E.g: When using PINN networks on a electrical circuit, one might use a variable resistance dependent on one of your network input parameters. When this dependency is given by an explicit equation it would be nice to, given the inputs, be able to use existing solvers as scipy's fsolve to get this value and use it further on in the network.

From my experience in order to be able to use solvers, tensors should be converted to numpy elements and used in combination with tf.numpy_function, this way I can use solvers as such, however it breaks the chain of gradients and TF can no longer find gradients for certain values and thus no longer train the network.

**System information**
- Tensorflow 2.3.0 (but using some TF1 functionalities from tf.compat.v1:
- No possibility to share existing code, but see below for small (hypothetical) case example. 

### start of code
from scipy.optimize import fsolve
Class PPINlayer(tf.keras.layers.Layer)
       def __init__(self):
               super(PPINlayer, self).__init__(**kwargs)

       def call(self, input, NN *args):
               NNoutput = NN(input)                # this is a NN already initialized somewhere else with trainable parameters
               output = self.f(NNoutput, input)
               return output                                # the output is then used later on to compare with labeled data as loss function for training
       def f(self, NNoutput, input)
             R0 = input[0]
             R1 = input[1]
             V0 = input[2]
             def R2_eq(R2, R0, R1):
                    return np.exp(R2/R1) * R1/R2 + R0**2
             R2 = fsolve(R2_eq, 0, args=(R0, R1))
             return (V0/R0 * R2)

Thanks in advance!
Cedric

**Comment by mohantym** [comment]
Hi @cevheck !
Please post this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . Thanks!

**Comment by cevheck** [comment]
Hi @mohantym!
I will do that, I'll close this issue then.

Thanks!
